src/utils/sidebar-links.tsx

```diff
@@ -0,0 +1,105 @@
+import { useMemo } from "react"
+import { useAtom, useAtomValue } from "jotai"
+import { useQuery } from "@tanstack/react-query"
+
+import { getCategoryList } from "./query-category"
+
+import { SidebarLink } from "../types/sidebar-link"
+import { createDashboardIdAtom } from "../store/create"
+import { siteAtom } from "../store/site"
+import { CustomIcon } from "../components/CustomIcon"
+import { Box, Divider } from "@mantine/core"
+
+const CATEGORY_ICONS: Record<string, string> = {
+  "Leadership Training": "iconoir:leaderboard-star",
+  "Technical Skills": "iconoir:code",
+  "Soft Skills": "iconoir:people-tag",
+  Compliance: "tabler:scale",
+  "Health & Wellness": "iconoir:health-shield",
+  Marketing: "iconoir:megaphone",
+}
+
+export function useSidebarLinks(): SidebarLink[] {
+  const site = useAtomValue(siteAtom)
+
+  const categoryQuery = useQuery({
+    queryKey: ["categories", site],
+    queryFn: () => getCategoryList(site),
+  })
+
+  const [, setDashboardId] = useAtom(createDashboardIdAtom)
+
+  return useMemo(() => {
+    const categories = categoryQuery.data ?? []
+
+    const categoryLinks: SidebarLink[] = categories.map((category) => ({
+      title: category.name,
+      to: `/admin/categories/${category.id}`,
+      icon: CATEGORY_ICONS[category.name],
+    }))
+
+    return [
+      {
+        title: "Products",
+        children: [
+          {
+            to: "/admin/products",
+            title: "All products",
+            icon: "iconoir:report-columns",
+          },
+          ...categoryLinks,
+        ],
+        defaultOpened: true,
+      },
+      {
+        component: () => (
+          <Box className="show-only-on-proficiency py-3">
+            <Divider
+              orientation="horizontal"
+              className="proficiency-sidebar-divider"
+            />
+          </Box>
+        ),
+      },
+      {
+        title: "Analytics",
+        children: [
+          {
+            to: "/admin/analytics",
+            title: "Dashboards",
+            icon: () => (
+              <CustomIcon
```

Comment:
it was really hard to understand why this wasn't rendered for other themes, and from scrolling a long PR I found that this is defined in the theme. I'm not sure if this is the best way to do this.

Maybe it's okay, but I was just confused why this icon isn't rendered for other themes. I wish it's structured in a way that you can put these custom icons in a theme-specific file that it'd be more obvious that these icons are only rendering on certin theme.

Reply:
That was indeed confusing - thanks for catching this! Fixed in https://github.com/metabase/shoppy/pull/91/commits/846b7791e0b9acfab0c4aab25c8998cdcbcb83bc by making the icons object site-specific (i.e. `{ icons: { proficiency: "foo" } }`)